src/lib/esnext.promise.d.ts

```diff
@@ -0,0 +1,17 @@
+interface PromiseWithResolvers<T> {
+    promise: Promise<T>;
+    resolve: (value: T | PromiseLike<T>) => void;

```

Comment:
AFAICT `value` should be `value?`, otherwise linter complains on `resolve()` about a missing argument of type "unknown".

Reply:
It can't be `value?` because then you could resolve _any_ promise with `undefined` accidentally by omitting the parameter.

I'm not sure how to do better other than to create two variants of `PromiseWithResolvers` and then overload `withResolvers` such that when `T` is undefined, you are allowed to have an optional parameter.

Reply:
FWIW these were copied from here to match the existing resolve and reject functions passed to the Promise constructor callback:

https://github.com/microsoft/TypeScript/blob/b7f691d9f13be987e445c7878a40190ca5808691/src/lib/es2015.promise.d.ts#L13

```ts
const p = new Promise((resolve) => {
    resolve(); // Expected 1 arguments, but got 0. Did you forget to include 'void' in your type argument to 'Promise'?(2794)
});
```

The suggested fix also works here, this typechecks fine:

```ts
const { resolve } = Promise.withResolvers<void>();
resolve();
```

Reply:
That's "the solution". And any other idea would be workaround. That's the official way of making `resolve` without arguments for `Promise`. You can't make `new Promise(resolve => resolve())` neither, you would need to provide `void` type parameter

Reply:
I see, thanks.

FWIW, I was trying to use these definitions in plain JS, not TS. I guess I'll just edit the files locally.